### ISSUE TITLE
Multiple locations with different step limit each

### ISSUE THREAD

**Issue by InflamedSebi** [AHAAAAAAA/PokemonGo-Map#3160]
Sometimes the shape of an hexagon doesn't fit your town/region.
wouldn't it be nice to have multiple locations with a specific step limit each?
and maybe one dynamic region that can be set via marker or geolocation.

## Current Behavior
for now you have to use multiple worker for different locations and split your accounts among them.

## Expected Behavior
there sould be a similar approach like multi accounting.
so on command line you can have multiple -l <some location> -l <another location> -st 3 -st 6
and if -fl is not set, the last of all those locations would be the moving one.

in config you could do similar to the accounts:
location: ["some loc", "another loc", "dynamic loc"]
step_limit: [4, 5, 3]

all the workers would scan the same location and switch to the next one as soon as the current one is done. And at last start all over again.

## Context
Actually this would allow to efficiently scan high priority areas and scan low priority areas just once in a while, because you could define some areas more often:
location: ["loc1", "loc2", "loc1", "loc3"]
step_limit: [4, 6, 4, 3]
loc1 is scanned twice as often than location 2 and 3

or:
location: ["whole town", "spot1", "spot2", "spot3", "spot1", "spot2", "spot3"]
step_limit: [10, 3, 3, 3, 3, 3, 3]

or use step_limit 1 sized locations somewhere around your town to update gyms that are out of reach normaly.

**Comment by shiona** [comment]
This can be handled with multiple instances of runserver.py.
`./runserver.py -u user1 -u user2 -p pass -l loc1 -st 6`
`./runserver.py -u user3 -p pass -l loc2 -st 3 -ns`

**Comment by InflamedSebi** [comment]
Ya, i know 

> for now you have to use multiple worker for different locations and split your accounts among them.

having it all in one would allow for still using sqlite and not need to setup mysql or similar.

**Comment by shiona** [comment]
Oh, I didn't know multiple runservers required a "real" database.

It would be easy enough to implement multiple locations, but all workers would first go through first location, then the second etc. until all locations would have been checked. Then they would start again. This wouldn't be good if you wanted to use this for prioritizing areas. 

Multiple areas with dedicated groups of workers would require major changes in the code and I don't see it as a priority feature.

**Comment by miaojs** [comment]
You can run. A few instances using SQLite.

**Comment by InflamedSebi** [comment]
SQLite is actually a single user database. It will work with more as long as they don't access it the exact same time.

> It would be easy enough to implement multiple locations, but all workers would first go through first location, then the second etc. until all locations would have been checked. Then they would start again. 

This is the intended behaviour and priorities are made by adding the exact same location multiple times to the list. There is no need to add extra priority groups or similar:
The workers would scan one location after the other on the list and due to our special location is on this list multiple times, it will be scanned multiple times. This is similar to "high priority". A location which is only once on this list will only be scanned once of course. This is similar to "low priority".

So priorities will automatically be there, even if it is kinda passive like that.

**Comment by shiona** [comment]
There's the slight problem that:
Let's say you have one large area and what you want to give priority to is right in center of that. So you cycle scanning the center and right after that start scanning from the center again and all the way to the edges of the large area, leaving the center without attention for just the same time you would normally.

**Comment by miaojs** [comment]
A sqlite3 db lock wouldn't cause an instance to discard data. It would wait until the timeout before erroring.

**Comment by neobithia** [comment]
I second this feature!
